### Exercise 1.5
Consider the following metric:
$$
dl^2 = du^2 + dv^2 + dw^2 - \left(\frac{3}{13}du + \frac{4}{13}dv + \frac{12}{13}dw\right)^2
$$

It appears that this is a three-dimensional metric in u, v, and w. This is not true! This
metric is in fact two-dimensional in disguise. Prove this by finding a transformation between
the “coordinates” u, v, and w and the Cartesian coordinates x and y, for
example, that reduces the metric to:

$$
dl^2 = dx^2 + dy^2
$$

#### Simplifly expression

In [1]:
var('du dv dw')
dl2 = du^2 + dv^2 + dw^2 - (3/13*du + 4/13*dv + 12/13*dw)^2
dl2 = dl2.expand()
dl2

160/169*du^2 - 24/169*du*dv + 153/169*dv^2 - 72/169*du*dw - 96/169*dv*dw + 25/169*dw^2

#### Extract coefficients and build metric matrix

In [33]:
coeff_du2 = dl2.coefficient(du^2)
coeff_dv2 = dl2.coefficient(dv^2)
coeff_dw2 = dl2.coefficient(dw^2)
coeff_dudv = dl2.coefficient(du*dv)
coeff_dudw = dl2.coefficient(du*dw)
coeff_dvdw = dl2.coefficient(dv*dw)

In [34]:
A = Matrix([
    [coeff_du2, coeff_dudv/2, coeff_dudw/2],
    [coeff_dudv/2, coeff_dv2, coeff_dvdw/2],
    [coeff_dudw/2, coeff_dvdw/2, coeff_dw2]
])
A

[160/169 -12/169 -36/169]
[-12/169 153/169 -48/169]
[-36/169 -48/169  25/169]

#### Diagonalize metric and check dimensions of new coordinate system
Confirm metric is two dimensional in disguise by checking eigenvalues such that

$$
dl^2 = dx^2 + dy^2
$$

In [35]:
eigenvalues = A.eigenvalues()
eigenvectors = A.eigenvectors_right()
eigenvalues

[1, 1, 0]

#### Calculate change of basis matrix and coordinate system transformations

In [36]:
Q = Matrix([eigenvectors[0][1][0], 
            eigenvectors[0][1][1], 
            eigenvectors[1][1][0]]).transpose() 
Q*Matrix([[1, 0, 0],[0, 1, 0],[0, 0, 0]])*Q.inverse() == A
Q.inverse()*A*Q

[1 0 0]
[0 1 0]
[0 0 0]

In [37]:
var('u v w')
uvw = vector([u , v, w])
transform = Q.inverse() * uvw
transform

(160/169*u - 12/169*v - 36/169*w, -12/169*u + 153/169*v - 48/169*w, 9/169*u + 12/169*v + 36/169*w)

$$
x = \frac{160}{169}u - \frac{12}{169}v - \frac{36}{169}w\newline
y = -\frac{12}{169}u + \frac{153}{169}v - \frac{48}{169}w\newline
z = \frac{9}{169}u + \frac{12}{169}v + \frac{36}{169}w
$$

#### Prove that the metric in the new coordinates is $$dl^2 = dx^2 + dy^2$$

In [30]:
dl2

160/169*du^2 - 24/169*du*dv + 153/169*dv^2 - 24/169*(3*du + 4*dv)*dw + 25/169*dw^2

In [27]:
dxyz = transform.subs(u=du, v=dv, w=dw)
(dxyz[0]^2 + dxyz[2]^2).full_simplify()

28721/28561*du^2 - 24/28561*du*dv + 2857/28561*dv^2 - 6/28561*(12*du - 2857*dv)*dw + 25713/28561*dw^2

In [46]:
ds2 = du^2 + 6* du * dv + 13 * dv^2
g = Matrix([[1, 3],[3, 13]])
eigenvec = g.eigenvectors_right()
P = Matrix([eigenvec[0][1][0], eigenvec[1][1][0]]).transpose()
P.inverse()*g*P
P.inverse() * vector([u, v])

(0.9472135954999579?*u - 0.2236067977499790?*v, 0.05278640450004206?*u + 0.2236067977499790?*v)